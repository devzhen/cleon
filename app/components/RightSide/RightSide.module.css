.rightSide {
  position: relative;
  display: grid;
  grid-template-rows: 59px 185px 1fr 205px;
  grid-template-columns: 1fr;
  height: calc(100vh - 72px);
  min-width: 1152px;
  min-height: 863px;
  border: 3px solid var(--color-board-background);
  margin-left: 3px;
  background-color: var(--color-board-background);
  z-index: 2;
}

.rightSide [data-role='drag-area'] {
  position: absolute;
}

.rightSide img {
  cursor: pointer;
}

.rightSide [data-role='header'] {
  display: flex;
  justify-content: center;
  align-items: center;
  user-select: none;
}

.rightSide [data-role='header'] span {
  margin-left: 10px;
  color: var(--color-header-right-side);
  font-size: 28px;
  text-transform: uppercase;
}

.rightSide [data-role='buttons-row'] {
  display: flex;
  justify-content: flex-end;
  align-items: flex-start;
}

.rightSide [data-role='banners-row'] div > img:nth-child(2) {
  margin-left: 10px;
}

.rightSide [data-role='buttons-row'] div:nth-child(2) > img:nth-child(2) {
  margin-left: 10px;
  margin-right: 10px;
}

.rightSide [data-role='banners-row'] {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
}

.rightSide [data-role='banners-row'] div {
  display: flex;
  align-items: center;
  width: content-fit;
}
