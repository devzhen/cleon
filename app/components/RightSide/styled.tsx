import styled from "styled-components";

export const Container = styled.div`
  position: relative;
  display: grid;
  grid-template-rows: 59px 185px 1fr 205px;
  grid-template-columns: 1fr;
  height: calc(100vh - 72px);
  min-width: 1152px;
  min-height: 863px;
  border: 3px solid ${(props) => props.theme.colors.notActiveBoard};
  margin-left: 3px;
  background-color: ${(props) => props.theme.colors.boardBackground};
  z-index: 2;

  img {
    cursor: pointer;
  }

  .drag-area {
    position: absolute;
  }
`;

export const Header = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  user-select: none;

  span {
    margin-left: 10px;
    color: ${(props) => props.theme.colors.headerRightSide};
    font-size: 28px;
    text-transform: uppercase;
  }
`;

export const ButtonsRow = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: start;

  div:nth-child(2) > img:nth-child(2) {
    margin-left: 10px;
    margin-right: 10px;
  }
`;

export const BannersRow = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px 10px 10px 10px;

  div {
    display: flex;
    align-items: center;
    width: content-fit;

    img:nth-child(2) {
      margin-left: 10px;
    }
  }
`;

export const FloatingWrapper = styled.div<{
  $paddingTop?: number;
}>`
  width: fit-content;
  height: fit-content;

  ${(props) => props.$paddingTop && `padding-top: ${props.$paddingTop}px`};
`;
