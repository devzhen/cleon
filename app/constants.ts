export const DEFAULT_COORDS = {
  top: 100,
  left: 220,
} as const;

export const BOARD_TYPE = {
  left: 'left',
  right: 'right',
} as const;