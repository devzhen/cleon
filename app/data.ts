import { v4 as uuid } from "uuid";

export const stickers = [
  {
    id: uuid(),
    top: 110,
    left: 230,
    createdAt: new Date().toISOString(),
    text: ''
  }
];
