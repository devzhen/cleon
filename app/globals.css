:root {
  --color-active-board: #eb6529;
  --color-not-active-board: #bfc3ca;
  --color-main-background: #f0cc46;
  --color-board-background: #e9ecf1;
  --color-header-right-side: #7b7b7b;
  --color-modal-background: #fff8c8;
  --color-text-area-background: #fffde6;
  --color-text-area-border: #dfe2e7;
  --color-text-area-color: #2e2e2e;
  --color-date-color: #979797;
}

html,
body {
  max-width: 100vw;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.main {
  display: grid;
  grid-template-rows: 72px 1fr;
  grid-template-columns: 638px 1fr;
  height: 100dvh;
  width: 100%;
}

.boardActive {
  border-color: var(--color-active-board) !important;
}

.boardNotActive {
  border-color: var(--color-not-active-board) !important;
}
