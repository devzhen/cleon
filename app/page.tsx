"use client";

import styled from "styled-components";

import Header from "./components/Header";
import LeftSide from "./components/LeftSide";

export default function Home() {
  return (
    <Main>
      <header>
        <Header />
      </header>
      <LeftSide />
    </Main>
  );
}

const Main = styled.main`
  display: grid;
  grid-template-rows: 72px 1fr;
  grid-template-columns: 638px 1fr;
  height: 100dvh;
  width: 100%;

  header {
    grid-column: 2 span;
    height: 66px;
    width: 100%;
  }
`;
