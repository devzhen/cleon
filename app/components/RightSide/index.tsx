import Image from "next/image";
import { useRef } from "react";

import { BOARD_TYPE } from "@/app/constants";
import useDraggableAreas from "@/app/hooks/useDraggableAreas";
import useSticker from "@/app/hooks/useSticker";
import getCoords from "@/app/utils/getCoords";
import isPointInPath from "@/app/utils/isPointInPath";

import ButtonsAddDelete from "../ButtonsAddDelete";
import ModalAddSticker from "../ModalAddSticker";
import StickerComponent from "../Sticker";

import { Container, Header, ButtonsRow, BannersRow } from "./styled";

export default function LeftSide() {
  // Current draggable sticker
  const currentDragElementRef = useRef<HTMLDivElement>();

  // Current draggable sticker's parent
  const currentDragElementParentRef = useRef<HTMLDivElement>();

  // Drag containers
  const dragContainerRef = useRef<HTMLDivElement>(null);
  const dragArea1Ref = useRef<HTMLDivElement>(null);
  const dragArea2Ref = useRef<HTMLDivElement>(null);
  const dragArea3Ref = useRef<HTMLDivElement>(null);
  const dragArea4Ref = useRef<HTMLDivElement>(null);
  const dragArea5Ref = useRef<HTMLDivElement>(null);

  // Banners, logos, images
  const headerRef = useRef<HTMLDivElement>(null);
  const buttonAddDeleteRef = useRef<HTMLDivElement>(null);
  const topImagesRef = useRef<HTMLImageElement>(null);
  const logoRef = useRef<HTMLImageElement>(null);
  const bottomLeftImagesRef = useRef<HTMLDivElement>(null);
  const bottomRightImagesRef = useRef<HTMLDivElement>(null);

  /**
   * Init draggable areas
   */
  const initDragAreas = () => {
    if (
      !dragContainerRef.current ||
      !dragArea1Ref.current ||
      !dragArea2Ref.current ||
      !dragArea3Ref.current ||
      !dragArea4Ref.current ||
      !dragArea5Ref.current ||
      !headerRef.current ||
      !logoRef.current ||
      !buttonAddDeleteRef.current ||
      !topImagesRef.current ||
      !bottomLeftImagesRef.current ||
      !bottomRightImagesRef.current
    ) {
      return;
    }

    const BORDER_WIDTH = 3;

    const dragContainerRefRect =
      dragContainerRef.current.getBoundingClientRect();
    const headerRect = headerRef.current.getBoundingClientRect();
    const topImagesRefRect = topImagesRef.current.getBoundingClientRect();
    const buttonAddDeleteRefRect =
      buttonAddDeleteRef.current.getBoundingClientRect();
    const bottomLeftImagesRefRect =
      bottomLeftImagesRef.current.getBoundingClientRect();
    const bottomRightImagesRefRect =
      bottomRightImagesRef.current.getBoundingClientRect();

    const dragArea1Rect = {
      left: 0,
      top: 0,
      width: logoRef.current.offsetLeft,
      height: bottomLeftImagesRef.current.offsetTop,
    };

    dragArea1Ref.current.style.top = `${dragArea1Rect.top}px`;
    dragArea1Ref.current.style.left = `${dragArea1Rect.left}px`;
    dragArea1Ref.current.style.width = `${dragArea1Rect.width}px`;
    dragArea1Ref.current.style.height = `${dragArea1Rect.height}px`;
    dragArea1Ref.current.style.background = "blue";
    dragArea1Ref.current.style.opacity = "0.5";

    const dragArea2Rect = {
      left: 0,
      top: headerRect.height,
      width: buttonAddDeleteRef.current.offsetLeft,
      height: bottomLeftImagesRef.current.offsetTop - headerRect.height,
    };

    dragArea2Ref.current.style.top = `${dragArea2Rect.top}px`;
    dragArea2Ref.current.style.left = `${dragArea2Rect.left}px`;
    dragArea2Ref.current.style.width = `${dragArea2Rect.width}px`;
    dragArea2Ref.current.style.height = `${dragArea2Rect.height}px`;
    dragArea2Ref.current.style.opacity = "0.5";

    const dragArea3Rect = {
      left: 0,
      top: buttonAddDeleteRef.current.offsetTop + buttonAddDeleteRefRect.height,
      width: topImagesRef.current.offsetLeft,
      height:
        bottomLeftImagesRef.current.offsetTop -
        headerRect.height -
        buttonAddDeleteRefRect.height,
    };

    dragArea3Ref.current.style.top = `${dragArea3Rect.top}px`;
    dragArea3Ref.current.style.left = `${dragArea3Rect.left}px`;
    dragArea3Ref.current.style.width = `${dragArea3Rect.width}px`;
    dragArea3Ref.current.style.height = `${dragArea3Rect.height}px`;
    dragArea3Ref.current.style.opacity = "0.5";

    const dragArea4Rect = {
      left: 0,
      top: topImagesRef.current.offsetTop + topImagesRefRect.height,
      width: dragContainerRefRect.width - BORDER_WIDTH * 2,
      height:
        bottomRightImagesRef.current.offsetTop -
        headerRect.height -
        topImagesRefRect.height,
    };

    dragArea4Ref.current.style.top = `${dragArea4Rect.top}px`;
    dragArea4Ref.current.style.left = `${dragArea4Rect.left}px`;
    dragArea4Ref.current.style.width = `${dragArea4Rect.width}px`;
    dragArea4Ref.current.style.height = `${dragArea4Rect.height}px`;
    dragArea4Ref.current.style.opacity = "0.5";

    const dragArea5Rect = {
      left:
        bottomLeftImagesRefRect.width + bottomLeftImagesRef.current.offsetLeft,
      top:
        bottomLeftImagesRef.current.offsetTop - bottomLeftImagesRefRect.height,
      width:
        bottomRightImagesRef.current.offsetLeft -
        bottomLeftImagesRefRect.width -
        bottomLeftImagesRef.current.offsetLeft,
      height: bottomRightImagesRefRect.height * 2,
    };

    dragArea5Ref.current.style.top = `${dragArea5Rect.top}px`;
    dragArea5Ref.current.style.left = `${dragArea5Rect.left}px`;
    dragArea5Ref.current.style.width = `${dragArea5Rect.width}px`;
    dragArea5Ref.current.style.height = `${dragArea5Rect.height}px`;
    dragArea5Ref.current.style.opacity = "0.5";
  };

  /**
   * Detect parent container
   */
  const detectParentContainer = () => {
    if (
      !dragContainerRef.current ||
      !dragArea1Ref.current ||
      !dragArea2Ref.current ||
      !dragArea3Ref.current ||
      !dragArea4Ref.current ||
      !dragArea5Ref.current ||
      !headerRef.current ||
      !logoRef.current ||
      !buttonAddDeleteRef.current ||
      !topImagesRef.current ||
      !bottomLeftImagesRef.current ||
      !bottomRightImagesRef.current
    ) {
      return;
    }

    const stickerCoords = getCoords(
      currentDragElementRef.current as HTMLDivElement
    );

    const dragArea1Coords = getCoords(dragArea1Ref.current);
    const dragArea2Coords = getCoords(dragArea2Ref.current);
    const dragArea3Coords = getCoords(dragArea3Ref.current);
    const dragArea4Coords = getCoords(dragArea4Ref.current);
    const dragArea5Coords = getCoords(dragArea5Ref.current);

    // The drag area 1
    if (currentDragElementParentRef.current === dragArea1Ref.current) {
      if (
        stickerCoords.top > dragArea1Coords.top &&
        isPointInPath(
          dragArea2Ref.current,
          stickerCoords.left,
          stickerCoords.top
        )
      ) {
        currentDragElementParentRef.current = dragArea2Ref.current;
      }
    }

    // The drag area 2
    if (currentDragElementParentRef.current === dragArea2Ref.current) {
      if (
        stickerCoords.top < dragArea2Coords.top &&
        isPointInPath(
          dragArea1Ref.current,
          stickerCoords.right,
          stickerCoords.top
        )
      ) {
        currentDragElementParentRef.current = dragArea1Ref.current;
      }

      if (
        stickerCoords.top >= dragArea3Coords.top &&
        isPointInPath(
          dragArea3Ref.current,
          stickerCoords.right,
          stickerCoords.top
        )
      ) {
        currentDragElementParentRef.current = dragArea3Ref.current;
      }
    }

    // The drag area 3
    if (currentDragElementParentRef.current === dragArea3Ref.current) {
      if (
        stickerCoords.top < dragArea3Coords.top &&
        isPointInPath(
          dragArea2Ref.current,
          stickerCoords.right,
          stickerCoords.top
        )
      ) {
        currentDragElementParentRef.current = dragArea2Ref.current;
      }

      if (
        isPointInPath(
          dragArea4Ref.current,
          stickerCoords.right,
          stickerCoords.top
        )
      ) {
        currentDragElementParentRef.current = dragArea4Ref.current;
      }
    }

    // The drag area 4
    if (currentDragElementParentRef.current === dragArea4Ref.current) {
      if (
        stickerCoords.top < dragArea4Coords.top &&
        isPointInPath(
          dragArea3Ref.current,
          stickerCoords.right,
          stickerCoords.top
        )
      ) {
        currentDragElementParentRef.current = dragArea3Ref.current;
      }

      if (
        isPointInPath(
          dragArea5Ref.current,
          stickerCoords.left,
          stickerCoords.top
        ) &&
        isPointInPath(
          dragArea5Ref.current,
          stickerCoords.right,
          stickerCoords.top
        ) &&
        isPointInPath(
          dragArea5Ref.current,
          stickerCoords.left,
          stickerCoords.bottom
        ) &&
        isPointInPath(
          dragArea5Ref.current,
          stickerCoords.right,
          stickerCoords.bottom
        )
      ) {
        currentDragElementParentRef.current = dragArea5Ref.current;
      }
    }

    // The drag area 5
    if (currentDragElementParentRef.current === dragArea5Ref.current) {
      if (
        stickerCoords.top < dragArea5Coords.top &&
        (!isPointInPath(
          dragArea5Ref.current,
          stickerCoords.left,
          stickerCoords.top
        ) ||
          !isPointInPath(
            dragArea5Ref.current,
            stickerCoords.right,
            stickerCoords.top
          ))
      ) {
        currentDragElementParentRef.current = dragArea4Ref.current;
      }
    }
  };

  /**
   * Correct a sticker's position relative to a parent
   */
  const correctPositionRelativeToParent = () => {
    if (
      !currentDragElementRef.current ||
      !currentDragElementParentRef.current
    ) {
      return;
    }

    const stickerCoords = getCoords(
      currentDragElementRef.current as HTMLDivElement
    );

    const parentCoords = getCoords(
      currentDragElementParentRef.current as HTMLDivElement
    );

    if (stickerCoords.left < parentCoords.left) {
      currentDragElementRef.current.style.left = `${currentDragElementParentRef.current.offsetLeft}px`;
    }

    if (stickerCoords.top < parentCoords.top) {
      currentDragElementRef.current.style.top = `${currentDragElementParentRef.current.offsetTop}px`;
    }

    if (stickerCoords.left + stickerCoords.width > parentCoords.right) {
      currentDragElementRef.current.style.left = `${
        currentDragElementParentRef.current.offsetLeft +
        parentCoords.width -
        stickerCoords.width
      }px`;
    }

    if (stickerCoords.top + stickerCoords.width > parentCoords.bottom) {
      currentDragElementRef.current.style.top = `${
        currentDragElementParentRef.current?.offsetTop +
        parentCoords.height -
        stickerCoords.height
      }px`;
    }
  };

  /**
   * Correct a sticker's position relative to a drag container
   */
  const correctPositionRelativeToDragContainer = () => {
    if (!currentDragElementRef.current || !dragContainerRef.current) {
      return;
    }

    const stickerCoords = getCoords(
      currentDragElementRef.current as HTMLDivElement
    );

    const dragContainerCoords = getCoords(
      dragContainerRef.current as HTMLDivElement
    );

    if (stickerCoords.left < dragContainerCoords.left) {
      currentDragElementRef.current.style.left = `${0}px`;
    }

    if (stickerCoords.top < dragContainerCoords.top) {
      currentDragElementRef.current.style.top = `${0}px`;
    }

    if (stickerCoords.left + stickerCoords.width > dragContainerCoords.right) {
      currentDragElementRef.current.style.left = `${
        dragContainerCoords.width - stickerCoords.width
      }px`;
    }

    if (stickerCoords.top + stickerCoords.width > dragContainerCoords.bottom) {
      currentDragElementRef.current.style.top = `${
        dragContainerCoords.height - stickerCoords.height
      }px`;
    }
  };

  /**
   * Use draggable areas hook
   */
  const { onMouseDown, onMouseMove } = useDraggableAreas({
    currentDragElementRef,
    dragContainerRef,
    onMountHandler: () => {
      initDragAreas();

      currentDragElementParentRef.current =
        dragArea2Ref.current as HTMLDivElement;
    },
    onMouseMoveHandler: () => {
      detectParentContainer();
      correctPositionRelativeToParent();
      correctPositionRelativeToDragContainer();
    },
  });

  /**
   * Use sticker hook
   */
  const {
    closeModalHandler,
    createSticker,
    editSticker,
    editedSticker,
    isModalOpen,
    setModalVisibility,
    stickers,
  } = useSticker({
    board: BOARD_TYPE.right,
  });

  return (
    <>
      <Container
        className="sticker-container"
        ref={dragContainerRef}
        onMouseMove={onMouseMove as VoidFunction}
      >
        <div className="drag-area drag-area-1" ref={dragArea1Ref} />
        <div className="drag-area drag-area-2" ref={dragArea2Ref} />
        <div className="drag-area drag-area-3" ref={dragArea3Ref} />
        <div className="drag-area drag-area-4" ref={dragArea4Ref} />
        <div className="drag-area drag-area-5" ref={dragArea5Ref} />
        <Header ref={headerRef}>
          <Image
            src="/scrin-3-graf.png"
            width={47}
            height={47}
            alt="Picture of the author"
            draggable={false}
            ref={logoRef}
          />
          <span>vie de l’uet</span>
        </Header>
        <ButtonsRow>
          <ButtonsAddDelete
            add={setModalVisibility(true)}
            remove={() => {}}
            marginRight={10}
            ref={buttonAddDeleteRef}
          />
          <div ref={topImagesRef}>
            <Image
              src="/scrin3-a3-right1.jpg"
              width={262}
              height={185}
              alt="Picture of the author"
              draggable={false}
              priority
            />
            <Image
              src="/scrin3-a3-right2.jpg"
              width={262}
              height={185}
              alt="Picture of the author"
              draggable={false}
              priority
            />
          </div>
        </ButtonsRow>
        <div />
        <BannersRow>
          <div ref={bottomLeftImagesRef}>
            <Image
              src="/scrin3-a3-right3.jpg"
              width={262}
              height={185}
              alt="Picture of the author"
              draggable={false}
              priority
            />
            <Image
              src="/scrin3-a3-right4.jpg"
              width={262}
              height={185}
              alt="Picture of the author"
              draggable={false}
              priority
            />
          </div>
          <div ref={bottomRightImagesRef}>
            <Image
              src="/scrin3-a4-right1.jpg"
              width={129}
              height={188}
              alt="Picture of the author"
              draggable={false}
              priority
            />
            <Image
              src="/scrin3-a4-right2.jpg"
              width={129}
              height={188}
              alt="Picture of the author"
              draggable={false}
              priority
            />
          </div>
        </BannersRow>
        {stickers.map((item) => (
          <StickerComponent
            onMouseDown={onMouseDown as VoidFunction}
            key={item.id}
            sticker={item}
            editSticker={editSticker(item)}
          />
        ))}
      </Container>
      {isModalOpen && (
        <ModalAddSticker
          closeModal={closeModalHandler}
          createSticker={createSticker}
          board={BOARD_TYPE.left}
          sticker={editedSticker}
        />
      )}
    </>
  );
}
