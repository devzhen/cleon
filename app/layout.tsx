import type { Metadata } from "next";
import { Roboto } from "next/font/google";

import "./globals.css";

import StyledComponentsRegistry from "../lib/registry";

const roboto = Roboto({
  weight: ["400", "700"],
  style: ["normal"],
  subsets: ["latin"],
  display: "swap",
});

export const metadata: Metadata = {
  title: "Cleon",
  description: "Cleon",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body className={roboto.className}>
        <StyledComponentsRegistry>{children}</StyledComponentsRegistry>
      </body>
    </html>
  );
}
