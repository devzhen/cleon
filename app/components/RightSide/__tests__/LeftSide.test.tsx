import { render } from '@testing-library/react';

import { firstId, stickers } from '@/app/data';

import RightSide from '..';

describe('RightSide tests', () => {
  const props = {
    initialStickers: {
      [firstId]: stickers[firstId],
    },
  };
  it('matches the snapshot', () => {
    const { container } = render(<RightSide {...props} />);

    expect(container).toMatchSnapshot();
  });
});
