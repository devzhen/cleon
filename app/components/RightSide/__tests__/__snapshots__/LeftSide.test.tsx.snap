// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightSide tests matches the snapshot 1`] = `
<div>
  <div
    class="rightSide sticker-container"
    data-board="right"
  >
    <div
      data-role="drag-area"
      style="top: 0px; left: 0px; width: 0px; height: 0px; opacity: 0.5;"
    />
    <div
      data-role="drag-area"
      style="top: 0px; left: 0px; width: 0px; height: 0px; opacity: 0.5;"
    />
    <div
      data-role="drag-area"
      style="top: 0px; left: 0px; width: 0px; height: 0px; opacity: 0.5;"
    />
    <div
      data-role="drag-area"
      style="top: 0px; left: 0px; width: -6px; height: 0px; opacity: 0.5;"
    />
    <div
      data-role="drag-area"
      style="top: 0px; left: 0px; width: 0px; height: 0px; opacity: 0.5;"
    />
    <div
      data-role="header"
    >
      <img
        alt="Picture of the author"
        data-nimg="1"
        decoding="async"
        draggable="false"
        height="47"
        loading="lazy"
        src="/_next/image?url=%2Fscrin-3-graf.png&w=96&q=75"
        srcset="/_next/image?url=%2Fscrin-3-graf.png&w=48&q=75 1x, /_next/image?url=%2Fscrin-3-graf.png&w=96&q=75 2x"
        style="color: transparent;"
        width="47"
      />
      <span>
        vie de lâ€™uet
      </span>
    </div>
    <div
      data-role="buttons-row"
    >
      <div
        class="container"
        style="margin-top: 0px; margin-right: 10px;"
      >
        <button>
          +
        </button>
        <button>
          -
        </button>
      </div>
      <div>
        <img
          alt="Picture of the author"
          data-nimg="1"
          decoding="async"
          draggable="false"
          fetchpriority="high"
          height="185"
          src="/_next/image?url=%2Fscrin3-a3-right1.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fscrin3-a3-right1.jpg&w=384&q=75 1x, /_next/image?url=%2Fscrin3-a3-right1.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="262"
        />
        <img
          alt="Picture of the author"
          data-nimg="1"
          decoding="async"
          draggable="false"
          fetchpriority="high"
          height="185"
          src="/_next/image?url=%2Fscrin3-a3-right2.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fscrin3-a3-right2.jpg&w=384&q=75 1x, /_next/image?url=%2Fscrin3-a3-right2.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="262"
        />
      </div>
    </div>
    <div />
    <div
      data-role="banners-row"
    >
      <div>
        <img
          alt="Picture of the author"
          data-nimg="1"
          decoding="async"
          draggable="false"
          fetchpriority="high"
          height="185"
          src="/_next/image?url=%2Fscrin3-a3-right3.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fscrin3-a3-right3.jpg&w=384&q=75 1x, /_next/image?url=%2Fscrin3-a3-right3.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="262"
        />
        <img
          alt="Picture of the author"
          data-nimg="1"
          decoding="async"
          draggable="false"
          fetchpriority="high"
          height="185"
          src="/_next/image?url=%2Fscrin3-a3-right4.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fscrin3-a3-right4.jpg&w=384&q=75 1x, /_next/image?url=%2Fscrin3-a3-right4.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="262"
        />
      </div>
      <div>
        <img
          alt="Picture of the author"
          data-nimg="1"
          decoding="async"
          draggable="false"
          fetchpriority="high"
          height="188"
          src="/_next/image?url=%2Fscrin3-a4-right1.jpg&w=384&q=75"
          srcset="/_next/image?url=%2Fscrin3-a4-right1.jpg&w=256&q=75 1x, /_next/image?url=%2Fscrin3-a4-right1.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="129"
        />
        <img
          alt="Picture of the author"
          data-nimg="1"
          decoding="async"
          draggable="false"
          fetchpriority="high"
          height="188"
          src="/_next/image?url=%2Fscrin3-a4-right2.jpg&w=384&q=75"
          srcset="/_next/image?url=%2Fscrin3-a4-right2.jpg&w=256&q=75 1x, /_next/image?url=%2Fscrin3-a4-right2.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="129"
        />
      </div>
    </div>
    <div
      class="container sticker"
      data-board="right"
      data-id="930c4351-09f2-4148-8b08-c39c9494fadb"
      role="presentation"
      style="top: 110px; left: 230px; z-index: 1;"
    >
      <span
        data-role="date"
      >
        02-03-2024 08:03:55 AM
      </span>
      <div
        data-role="content-wrapper"
      >
        <span>
          Draggable element 1
        </span>
      </div>
      <div
        data-role="image-wrapper"
      >
        <img
          alt="Picture of the author"
          data-nimg="1"
          decoding="async"
          draggable="false"
          height="13"
          loading="lazy"
          src="/_next/image?url=%2Freg.png&w=32&q=75"
          srcset="/_next/image?url=%2Freg.png&w=16&q=75 1x, /_next/image?url=%2Freg.png&w=32&q=75 2x"
          style="color: transparent;"
          width="13"
        />
      </div>
    </div>
  </div>
</div>
`;
